pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/MaxAggregationFunction.java

```diff
@@ -28,13 +30,23 @@
 import org.apache.pinot.core.query.aggregation.groupby.DoubleGroupByResultHolder;
 import org.apache.pinot.core.query.aggregation.groupby.GroupByResultHolder;
 import org.apache.pinot.segment.spi.AggregationFunctionType;
+import org.roaringbitmap.RoaringBitmap;
 
 
 public class MaxAggregationFunction extends BaseSingleInputAggregationFunction<Double, Double> {
   private static final double DEFAULT_INITIAL_VALUE = Double.NEGATIVE_INFINITY;
+  private final boolean _nullHandlingEnabled;
+
+  private final Set<Integer> _groupKeysWithNonNullValue;
```

Comment:
This requires extra lookup.
Instead, we can use object based result holder when null handling is enabled. We need it anyway in order to extract the aggregation result (no group-by) which is not properly handled right now. Same for `MIN` and `SUM`

Reply:
good suggestion